### PR TITLE
Flatten local struct in dpdk

### PR DESCRIPTION
**Pull request by kamleshbhalui**
flatten local struct  into metadata struct

### CODE REVIEW

File: backends/dpdk/dpdkArch.cpp [hunk 1]
```diff
@@ -936,6 +936,16 @@ const IR::Node *CollectLocalVariables::postorder(IR::Type_Struct *s) {
                 LOG3("Variable: " << dv << std::endl <<
                      " type: " << type << std::endl <<
                      " already added to: " << structure->header_type);
+            } else if(type->is<IR::Type_Struct>()) {
```
Comment (usha1830) [inline backends/dpdk/dpdkArch.cpp:939]: 939-940 can be combined into 
else if (auto strct = ...)

File: backends/dpdk/dpdkArch.cpp [hunk 2]
```diff
@@ -984,6 +994,16 @@ CollectLocalVariables::postorder(IR::PathExpression *p) {
     return p;
 }
 
+const IR::Node *
+CollectLocalVariables::postorder(IR::Member *m) {
+    auto expr = m->expr->to<IR::Member>();
+    if (expr && expr->expr->toString() == "m")
```
Comment (usha1830) [inline backends/dpdk/dpdkArch.cpp:1000]: A single line comment saying this is for flattening nested struct would be helpful here

**Review by usha1830 — CHANGES_REQUESTED** [review]
The test looks overly complicated. It would be nice to have a simpler test without vxlan encap/decap and as few actions as possible.

**Review by usha1830 — APPROVED** [review]
Looks fine.

File: testdata/p4_16_samples/psa-dpdk-flatten-local-struct.p4 [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+struct headers_t {
+    ethernet_t ethernet;
+    ipv4_t ipv4;
+    ethernet_t outer_ethernet;
+    ipv4_t outer_ipv4;
+}
+
+struct host_info_rx_bytes_t2
+
+{ bit<16> flex_0; bit<16> flex_1; bit<16> flex_2; bit<16> flex_3; bit<16> flex_4; bit<16> flex_5;}
```
Comment (usha1830) [inline testdata/p4_16_samples/psa-dpdk-flatten-local-struct.p4:37]: indentation looks weird. Having the fields on separate lines would be better.

File: testdata/p4_16_samples/psa-dpdk-flatten-local-struct.p4 [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        	host_info_rx_bytes.flex_up.flex_up1.flex_0 =  field;
+	else
+		host_info_rx_bytes.flex_up.flex_up1.flex_0 = field1;
+
+        headers.outer_ethernet.ether_type = host_info_rx_bytes.flex_up.flex_up1.flex_0;
+    }
+    action drop(){
+        ostd.egress_port = (PortId_t)4;
+    }
+    table vxlan {
```
Comment (usha1830) [inline testdata/p4_16_samples/psa-dpdk-flatten-local-struct.p4:77]: table name can be changed to something generic

**Review by usha1830 — APPROVED** [review]

**Review by psivanup — APPROVED** [review]

**Pull request closed without merging**